validator/src/journal/publisher.rs

```diff
@@ -173,15 +173,15 @@ impl SyncBlockPublisher {
             .pending_batches
             .rebuild(Some(committed_batches), Some(uncommitted_batches));
 
-        if let Some(block_id) = previous_block_id {
-            self.initialize_block(state, &block_id)
+        if let Some(prev) = previous_block_id {
```

Comment:
The previous_block_id variable should be renamed to reflect that the thing it points to is a Block and not a block id.